pilot/pkg/model/push_context.go

```diff
@@ -1707,12 +1707,21 @@ func (ps *PushContext) BestEffortInferServiceMTLSMode(service *Service, port *Po
 		return MTLSUnknown
 	}
 
-	// First , check mTLS settings from beta policy (i.e PeerAuthentication) at namespace / mesh level.
+	// 1. Check service instances' tls mode, mainly used for headless service.
+	if service.Resolution != ClientSideLB {
```

Comment:
Why do we do this for DNSLB?

Reply:
For service entry, the resolution can be DNSLB and with endpoints(wle) specified

Reply:
how is dns type different from eds in this code path? seems like it should be pretty much the same? Both are getting endpoints from an external source. Headless is the only unique one which does plain passthrough

Reply:
actually why do we even send headless as passthrough? We know all the endpoints... seems like it could be eds

Reply:
The only difference is: for dns type, the lb endpoints are inlined within cluster. 
You are right, then we can handle DNSLB same as ClientSideLB here.